Tests/Integration/CounterIntegrationTests.swift

```diff
@@ -0,0 +1,198 @@
+
+        XCTAssert(age!.value == 6)
+
+        var result = await doc.toSortedJSON()
+        XCTAssert(result == "{\"age\":6,\"length\":13}")
+
+        await doc.update { root in
+            do {
+                try (root.age as! JSONCounter<Int32>).increase(value: Int32(1)).increase(value: Int32(1))
+                try (root.length as! JSONCounter<Int64>).increase(value: Int64(3)).increase(value: Int64(1))
+            } catch {}
+        }
+
+        try await Task.sleep(nanoseconds: 1_000_000_000)
+
+        result = await doc.toSortedJSON()
+        XCTAssert(result == "{\"age\":8,\"length\":17}")
+
+        let expectation = XCTestExpectation(description: "failed Test")
+        var failedTestResult = false
+
+        await doc.update { root in
+            do {
+                try (root.age as! JSONCounter<Int32>).increase(value: Int64(1))
+            } catch {
+                failedTestResult = true
+                expectation.fulfill()
+            }
+        }
+
+        wait(for: [expectation], timeout: 1)
+
+        XCTAssert(failedTestResult)
+    }
+
+    func skip_test_can_sync_counter() async throws {
+        let options = ClientOptions()
+        let docKey = "\(self.description)-\(Date().description)"
+
+        do {
+            self.c1 = try Client(rpcAddress: self.rpcAddress, options: options)
+            self.c2 = try Client(rpcAddress: self.rpcAddress, options: options)
+        } catch {
+            XCTFail(error.localizedDescription)
+            return
+        }
+
+        self.d1 = Document(key: docKey)
+        self.d2 = Document(key: docKey)
+
+        try await self.c1.activate()
+        try await self.c2.activate()
+
+        try await self.c1.attach(self.d1, false)
+        try await self.c2.attach(self.d2, false)
+
+        try await Task.sleep(nanoseconds: 1_000_000_000)
+
+        await self.d1.update { root in
+            root.age = JSONCounter(value: Int32(1))
+        }
+
+        await self.d2.update { root in
+            root.length = JSONCounter(value: Int64(10))
+        }
+
+        try await Task.sleep(nanoseconds: 1_000_000_000)
+
+        var result1 = await self.d1.toSortedJSON()
+        var result2 = await self.d2.toSortedJSON()
+
+        XCTAssert(result1 == result2)
+
+        await self.d1.update { root in
+            do {
+                try (root.age as! JSONCounter<Int32>).increase(value: Int32(5))
+                try (root.length as! JSONCounter<Int64>).increase(value: Int64(3))
+            } catch {}
+        }
+
+        try await Task.sleep(nanoseconds: 1_000_000_000)
+
+        result1 = await self.d1.toSortedJSON()
+        result2 = await self.d2.toSortedJSON()
+
+        XCTAssert(result1 == result2)
+
+        try await self.c1.detach(self.d1)
+        try await self.c2.detach(self.d2)
+
+        try await self.c1.deactivate()
+        try await self.c2.deactivate()
+    }
+
+    func skip_test_can_sync_counter_with_array() async throws {
```

Comment:
To run the tests when a PR is created, remove skip_ from the method name.

Reply:
Nice!